### PR TITLE
doc: document sum type smart cast

### PR DESCRIPTION
**Pull request by ntrel**
~~Also fix doc CI embedding example.~~

### CODE REVIEW

**Comment by ntrel** [comment]
It's great that we can use CI to check complete code examples in the docs, but how do we suppress that when an example is an incomplete continuation of a previous example? Here it's inappropriate to repeat all the declarations for each of the two follow up examples. It's also important to explain a block of code before moving on to more advanced code. We don't want one huge example.

**Comment by spaceface777** [comment]
There's a comment at the top of the docs, that explains the different CI examples options; check-syntax may be good here, for example.

**Comment by ntrel** [comment]
@spaceface777 thanks, this is now green

File: doc/docs.md [hunk 1]
```diff
@@ -1643,16 +1645,46 @@ fn main() {
...
+        println('bad weather!')
+    }
+}
+```
+`w` has type `Mars` inside the body of the `if` statement. This is 
+known as *flow-sensitive typing*. You can also specify a variable name:
+
+```v oksyntax
+if w is Mars as mars {
+    assert typeof(w).name == 'World'
```
Comment (dumblob) [inline doc/docs.md:1666]: Shouldn't this be `assert typeof( mars ).name == 'Mars'`?

Comment (dumblob) [reply doc/docs.md:1666]: Ah, I see, you didn't want to demonstrate anything about `mars` but the fact, that `w` keeps its type. Ok, then it's fine :wink:.

**Pull request merged**